### DISCUSSION TITLE
Being able to set Nginx version

### DISCUSSION THREAD

**Discussion by khromov** [caprover/caprover#1859] [Ideas]
As far as I can tell, CapRover uses `nginx:1` [by default](https://github.com/caprover/caprover/blob/29638db990bfc772df595a292097e91e25f7eb67/src/utils/CaptainConstants.ts#L44) and occasionally updates the nginx version. (Although it's not clear to me when that happens.)

To avoid surprises in production environments, it would be cool if there was a setting in the admin panel where we could lock the nginx version, for example we could set it to `nginx:1.25` and be sure that the Nginx version won't automatically bump up and perhaps break our config (although in a perfect semver world that shouldn't happen.)

**Comment by githubsaturn** [comment]
Good idea! Just for the sake of transparency, "Being able to set Nginx version" is already possible. You just need to [override the relevant config](https://caprover.com/docs/troubleshooting.html#customize-config-settings). 

However, setting a fix version is probably a good idea. Feel free to create a PR for it! It's a fairly small change.

**Reply by githubsaturn** [reply]
Fixed here: https://github.com/caprover/caprover/commit/80cb3a3a90ba5840c020a7e3ddaf00cc19b239b4

**Reply by khromov** [reply]
Awesome, thanks for a quick fix! I noticed that I'm currently on Nginx 1.25. Will it be automatically downgraded at some point or during the next upgrade of Cap?

**Reply by githubsaturn** [reply]
It'll be downgraded when you upgrade to the next version of CapRover. The reason I set it to 1.24 is that it's the stable version.
